coil-base/src/main/java/coil/memory/MemoryCache.kt

```diff
@@ -43,14 +49,80 @@ internal interface MemoryCache {
     fun maxSize(): Int
 
     /** Remove the value referenced by [key] from this cache if it is present. */
-    fun invalidate(key: String)
+    fun invalidate(key: Key)
 
     /** Remove all values from this cache. */
     fun clearMemory()
 
     /** @see ComponentCallbacks2.onTrimMemory */
     fun trimMemory(level: Int)
 
+    /** Cache key for [MemoryCache] and [WeakMemoryCache]. */
+    class Key {
```

Comment:
No data class, unfortunately, since we need to be able to control when `size` is null.